projects/packages/search/src/initializers/class-initializer.php

```diff
@@ -7,14 +7,178 @@
 
 namespace Automattic\Jetpack\Search;
 
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
 use WP_Error;
-
 /**
  * Base class for the initializer pattern.
  */
-abstract class Initializer {
+class Initializer {
+	/**
+	 * Store the instance.
+	 *
+	 * @var Initializer
+	 */
+	protected static $instance;
+
+	/**
+	 * WPCOM blog ID
+	 *
+	 * @var int
+	 */
+	protected $blog_id;
+
+	/**
+	 * Initialize and get instance
+	 */
+	public static function instance() {
+		if ( is_null( static::$instance ) ) {
+			static::$instance = new static();
+			static::$instance->init();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Initialize the search package.
+	 */
+	protected function init() {
+		// Set up package version hook.
+		add_filter( 'jetpack_package_versions', __NAMESPACE__ . '\Package::send_version_to_tracker' );
+
+		if ( ! apply_filters( 'jetpack_search_initialize', true ) ) {
+			do_action( 'jetpack_search_abort', 'jetpack_search_initialize_filter', null );
+			return;
+		}
+
+		$this->init_before_connection();
+
+		// Check whether Jetpack Search should be initialized in the first place .
+		if ( ! $this->is_connected() || ! $this->is_search_supported() ) {
+			/**
+			 * Fires when the Jetpack Search fails and would fallback to MySQL.
+			 *
+			 * @since Jetpack 7.9.0
+			 * @param string $reason Reason for Search fallback.
+			 * @param mixed  $data   Data associated with the request, such as attempted search parameters.
+			 */
+			do_action( 'jetpack_search_abort', 'inactive', null );
+			return;
+		}
+
+		$this->blog_id = Helper::get_wpcom_site_id();
+		if ( ! $this->blog_id ) {
+			do_action( 'jetpack_search_abort', 'no_blog_id', null );
+			return;
+		}
+
+		$this->init_search_package();
+
+		// Fired when plugin ready.
```

Comment:
Could you expand that docblock?

```suggestion
		/**
		 * Fires when the Jetpack Search package has been initialized.
		 *
		 * @since $$next-version$$
		 */
```

I'm bugging you with those hook documentation updates because all hooks are documented on developer.jetpack.com, based on those inline comments, so I think it's nice to keep those as clear as possible.

Reply:
Hi @jeherve thanks for the feedback 👍  this is not bugging at all. Glad to know and will do that in the future. addressed in 737a6de7a5.